korok/korok.py

```diff
@@ -51,7 +81,89 @@ def fit(
             metric=Metric.COSINE,
             **kwargs,
         )
-        return cls(encoder=encoder, vicinity=vicinity, reranker=reranker)
+
+        # Add bm25s if alpha > 0
+        if cls.is_hybrid_available() and alpha < 1.0:
+            global bm25s
+            import bm25s
+
+            bm25 = bm25s.BM25()
+            tokens = bm25s.tokenize(texts, stopwords=stopwords)
+            bm25.index(tokens)
+        else:
+            bm25 = None
+
+        return cls(encoder=encoder, vicinity=vicinity, reranker=reranker, bm25=bm25, alpha=alpha, corpus=texts)
+
+    def _normalize_scores(self, scores: np.ndarray) -> np.ndarray:
+        """Normalize the scores to be between 0 and 1."""
+        # Get the min and max scores for each document
+        min_score = np.min(scores, axis=-1).reshape(-1, 1)
+        max_score = np.max(scores, axis=-1).reshape(-1, 1)
+
+        # Normalize the scores
+        denom = max_score - min_score
+        numerator = scores - min_score
+
+        return numerator / denom
+
+    def _split_vicinity_results(self, results: List[List[Tuple[str, float]]]) -> List[List[Tuple[str, float]]]:
```

Comment:
I think the return type is incorrect. Did mypy flag this?

Reply:
Oh, yes, it didn't get flagged for some reason; I will fix this! 

Does mypy need to be run separately? I did run ruff check and ruff format on the files once...